File: src/test/java/AccountTests.java
import com.dambra.paul.moduluschecker.Account.BankAccount;

import org.junit.Test;

import java.util.stream.Stream;

import static org.hamcrest.CoreMatchers.is;
import static org.hamcrest.MatcherAssert.assertThat;
import static org.hamcrest.core.IsEqual.equalTo;

public class AccountTests {
    @Test
    public void CanGetSortCodeAndAccountNumberForModulusChecks() {
        String sc = "012345";
        String an = "01234567";
        BankAccount account = new BankAccount(sc, an);
        Stream<Integer> expected = Stream.of(0, 1, 2, 3, 4, 5, 0, 1, 2, 3, 4, 5, 6, 7);
        Assert.thatStreamEquals(account.allDigits(), expected);
    }

    @Test
    public void CanUseAccountNotationToGetNumbersFromAccount() {
        String sc = "654321";
        String an = "98765432";
        BankAccount account = new BankAccount(sc, an);

        assertThat(account.getNumberAt(BankAccount.U), is(equalTo(6)));
        assertThat(account.getNumberAt(BankAccount.V), is(equalTo(5)));
        assertThat(account.getNumberAt(BankAccount.W), is(equalTo(4)));
        assertThat(account.getNumberAt(BankAccount.X), is(equalTo(3)));
        assertThat(account.getNumberAt(BankAccount.Y), is(equalTo(2)));
        assertThat(account.getNumberAt(BankAccount.Z), is(equalTo(1)));
        assertThat(account.getNumberAt(BankAccount.A), is(equalTo(9)));
        assertThat(account.getNumberAt(BankAccount.B), is(equalTo(8)));
        assertThat(account.getNumberAt(BankAccount.C), is(equalTo(7)));
        assertThat(account.getNumberAt(BankAccount.D), is(equalTo(6)));
        assertThat(account.getNumberAt(BankAccount.E), is(equalTo(5)));
        assertThat(account.getNumberAt(BankAccount.F), is(equalTo(4)));
        assertThat(account.getNumberAt(BankAccount.G), is(equalTo(3)));
        assertThat(account.getNumberAt(BankAccount.H), is(equalTo(2)));
    }

    @Test
    public void CanAvoidZeroiseForExceptionSeven() {
        String sc = "654321";
        String an = "98765432";
        BankAccount account = new BankAccount(sc, an).zeroiseUToB();

        assertThat(account.getNumberAt(BankAccount.U), is(equalTo(6)));
        assertThat(account.getNumberAt(BankAccount.V), is(equalTo(5)));
        assertThat(account.getNumberAt(BankAccount.W), is(equalTo(4)));
        assertThat(account.getNumberAt(BankAccount.X), is(equalTo(3)));
        assertThat(account.getNumberAt(BankAccount.Y), is(equalTo(2)));
        assertThat(account.getNumberAt(BankAccount.Z), is(equalTo(1)));
        assertThat(account.getNumberAt(BankAccount.A), is(equalTo(9)));
        assertThat(account.getNumberAt(BankAccount.B), is(equalTo(8)));
        assertThat(account.getNumberAt(BankAccount.C), is(equalTo(7)));
        assertThat(account.getNumberAt(BankAccount.D), is(equalTo(6)));
        assertThat(account.getNumberAt(BankAccount.E), is(equalTo(5)));
        assertThat(account.getNumberAt(BankAccount.F), is(equalTo(4)));
        assertThat(account.getNumberAt(BankAccount.G), is(equalTo(3)));
        assertThat(account.getNumberAt(BankAccount.H), is(equalTo(2)));
    }

    @Test
    public void CanZeroiseForExceptionSeven() {
        String sc = "654321";
        String an = "98765492";
        BankAccount account = new BankAccount(sc, an).zeroiseUToB();

        assertThat(account.getNumberAt(BankAccount.U), is(equalTo(0)));
        assertThat(account.getNumberAt(BankAccount.V), is(equalTo(0)));
        assertThat(account.getNumberAt(BankAccount.W), is(equalTo(0)));
        assertThat(account.getNumberAt(BankAccount.X), is(equalTo(0)));
        assertThat(account.getNumberAt(BankAccount.Y), is(equalTo(0)));
        assertThat(account.getNumberAt(BankAccount.Z), is(equalTo(0)));
        assertThat(account.getNumberAt(BankAccount.A), is(equalTo(0)));
        assertThat(account.getNumberAt(BankAccount.B), is(equalTo(0)));
        assertThat(account.getNumberAt(BankAccount.C), is(equalTo(7)));
        assertThat(account.getNumberAt(BankAccount.D), is(equalTo(6)));
        assertThat(account.getNumberAt(BankAccount.E), is(equalTo(5)));
        assertThat(account.getNumberAt(BankAccount.F), is(equalTo(4)));
        assertThat(account.getNumberAt(BankAccount.G), is(equalTo(9)));
        assertThat(account.getNumberAt(BankAccount.H), is(equalTo(2)));
    }

}

File: src/test/java/AccountTests.kt


import com.github.pauldambra.moduluschecker.account.BankAccount
import com.natpryce.hamkrest.assertion.assert
import com.natpryce.hamkrest.equalTo
import org.junit.Test

class AccountTests {
    @Test
    fun canGetSortCodeAndAccountNumberForModulusChecks() {
        val sc = "012345"
        val an = "01234567"
        val account = BankAccount(sc, an)
        val expected = listOf(0, 1, 2, 3, 4, 5, 0, 1, 2, 3, 4, 5, 6, 7)
        assert.that(account.allDigits(), equalTo(expected))
    }

    @Test
    fun canUseAccountNotationToGetNumbersFromAccount() {
        val sc = "654321"
        val an = "98765432"
        val account = BankAccount(sc, an)

        assert.that(account.getNumberAt(BankAccount.U), equalTo(6))
        assert.that(account.getNumberAt(BankAccount.V), equalTo(5))
        assert.that(account.getNumberAt(BankAccount.W), equalTo(4))
        assert.that(account.getNumberAt(BankAccount.X), equalTo(3))
        assert.that(account.getNumberAt(BankAccount.Y), equalTo(2))
        assert.that(account.getNumberAt(BankAccount.Z), equalTo(1))
        assert.that(account.getNumberAt(BankAccount.A), equalTo(9))
        assert.that(account.getNumberAt(BankAccount.B), equalTo(8))
        assert.that(account.getNumberAt(BankAccount.C), equalTo(7))
        assert.that(account.getNumberAt(BankAccount.D), equalTo(6))
        assert.that(account.getNumberAt(BankAccount.E), equalTo(5))
        assert.that(account.getNumberAt(BankAccount.F), equalTo(4))
        assert.that(account.getNumberAt(BankAccount.G), equalTo(3))
        assert.that(account.getNumberAt(BankAccount.H), equalTo(2))
    }

    @Test
    fun canAvoidZeroiseForExceptionSeven() {
        val sc = "654321"
        val an = "98765432"
        val account = BankAccount(sc, an).zeroiseUToB()

        assert.that(account.getNumberAt(BankAccount.U), equalTo(6))
        assert.that(account.getNumberAt(BankAccount.V), equalTo(5))
        assert.that(account.getNumberAt(BankAccount.W), equalTo(4))
        assert.that(account.getNumberAt(BankAccount.X), equalTo(3))
        assert.that(account.getNumberAt(BankAccount.Y), equalTo(2))
        assert.that(account.getNumberAt(BankAccount.Z), equalTo(1))
        assert.that(account.getNumberAt(BankAccount.A), equalTo(9))
        assert.that(account.getNumberAt(BankAccount.B), equalTo(8))
        assert.that(account.getNumberAt(BankAccount.C), equalTo(7))
        assert.that(account.getNumberAt(BankAccount.D), equalTo(6))
        assert.that(account.getNumberAt(BankAccount.E), equalTo(5))
        assert.that(account.getNumberAt(BankAccount.F), equalTo(4))
        assert.that(account.getNumberAt(BankAccount.G), equalTo(3))
        assert.that(account.getNumberAt(BankAccount.H), equalTo(2))
    }

    @Test
    fun canZeroiseForExceptionSeven() {
        val sc = "654321"
        val an = "98765492"
        val account = BankAccount(sc, an).zeroiseUToB()

        assert.that(account.getNumberAt(BankAccount.U), equalTo(0))
        assert.that(account.getNumberAt(BankAccount.V), equalTo(0))
        assert.that(account.getNumberAt(BankAccount.W), equalTo(0))
        assert.that(account.getNumberAt(BankAccount.X), equalTo(0))
        assert.that(account.getNumberAt(BankAccount.Y), equalTo(0))
        assert.that(account.getNumberAt(BankAccount.Z), equalTo(0))
        assert.that(account.getNumberAt(BankAccount.A), equalTo(0))
        assert.that(account.getNumberAt(BankAccount.B), equalTo(0))
        assert.that(account.getNumberAt(BankAccount.C), equalTo(7))
        assert.that(account.getNumberAt(BankAccount.D), equalTo(6))
        assert.that(account.getNumberAt(BankAccount.E), equalTo(5))
        assert.that(account.getNumberAt(BankAccount.F), equalTo(4))
        assert.that(account.getNumberAt(BankAccount.G), equalTo(9))
        assert.that(account.getNumberAt(BankAccount.H), equalTo(2))
    }

}
